app/services/task_service/gpt_generate_tests.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+module TaskService
+  class GptGenerateTests < ServiceBase
+    def initialize(task:)
+      super()
+      @task = task
+      @message = task.description
+      @language = @task.programming_language&.language
+      raise MissingLanguageError if @language.blank?
+    end
+
+    def execute
+      programming_language = @task.programming_language
+      programming_language.validation_context = :gpt
+
+      file_extension = programming_language.file_extension
+      testing_framework_name = programming_language.testing_framework
+      framework_version = programming_language.framework_version
+
+      file_content = gpt_response
+      test_file = TaskFile.new(content: file_content, name: "test.#{file_extension}", used_by_grader: true, visible: false,
```

Comment:
Wouldn't it be possible to outsource all those decisions to the AI as well? I could imagine modifying the prompt so that it also returns a fitting filename (with extension).

Reply:
I discussed this with Karol and there are many challenges to automating the correct filename creation. I think this should be resolved for the current PR and this challenge can be tackled in a future PR. 

For now, instructors can rename the test file accordingly.

Reply:
Okay, we can proceed with this approach. In this case, wouldn't it be suitable to suggest users to double check (and potentially correct) the file name chosen (or the test in general)? This could be integrated into the success message, I think.

Reply:
Yes, I can add that to the success message.

One extra future work thing I don't think should be too hard to add is the possibility for users to input their own API key for the feature to appear. not sure how I can proceed with doing this? Because currently, the feature is hidden for everyone without a key.

Reply:
Does this make sense:

successful_generation: Unit test generated successfully. Please check the generated test and provide the appropriate filename to it

successful_generation: Unit-Test erfolgreich generiert. Bitte prüfen Sie den generierten Test und geben Sie ihm den entsprechenden Dateinamen

Reply:
> One extra future work thing I don't think should be too hard to add is the possibility for users to input their own API key for the feature to appear. not sure how I can proceed with doing this? Because currently, the feature is hidden for everyone without a key.

Yes, that's a good future work. For now, and for this PR, I would suggest to proceed with the global key. Then, we can ship small releases and make a follow-up PR.

Adding the API key per user would probably involve another database column per user and a simple edit view as part of the profile. Then, we need to adapt the corresponding policy and use this user-specific key.

<hr>
Sounds good! For the German translation, I got a minor suggestion. Based on that, I updated the English locale as well, but this is optional (we can keep your translation there, too).

```yml
successful_generation: Unit test generated successfully. Please check the generated test and assign an appropriate filename

successful_generation: Unit-Test erfolgreich generiert. Bitte überprüfen Sie den generierten Test und vergeben Sie einen passenden Dateinamen
```